runtime/lua/vim/lsp/completion.lua

```diff
@@ -441,7 +445,7 @@ local function trigger(bufnr, clients)
   reset_timer()
   Context:cancel_pending()
 
-  if tonumber(vim.fn.pumvisible()) == 1 and not Context.isIncomplete then
+  if tonumber(vim.fn.pumvisible()) == 1 then
```

Comment:
```suggestion
  if tonumber(vim.fn.pumvisible()) == 1 and not Context.isIncomplete then
```

This looks unrelated?

Reply:
if keep there , continuing input will clean pum `wp.` then input w. pum is cleaned

Reply:
But isn't removing it breaking the functionality to fetch more results if the initial results were incomplete?
Is there some state hiccup that causes isIncomplete to have the wrong value?

Reply:
Maybe worth reporting upstream with clangd? In the meantime we could instead guard the `vim.fn.complete(start_col, matches)` and call it only if there are matches given that we rely on the pum mechanism to do the filtering we don't need a response to "wipe out" any shown candidates.

Reply:
Wait, let me research it.

Reply:
Resolved. and another question here is `Context.isIncomplete` should bind with client ? if one client return isIncomplete false do we want still send with all clients ?

Reply:
> (e.g. remove previous results for client-2, add new results for client-2, but keep results from client-1)

Can do in C side I prefer to do a soft delete here, that is, set an unavailable flag in the compl_T structure. Otherwise, there will be some memory allocation efficiency issues.

> Independent of that, do I understand it right that this PR is otherwise good to go?

I changed isInComplete to a table storing clientid. Do I need to revert to the original state? So far no tests have been broken. But maybe that means no catch.